app/src/main/java/com/orange/ods/app/ui/components/buttons/ComponentButtons.kt

```diff
@@ -63,32 +62,31 @@ fun ComponentButtons(variant: Variant) {
                 when (variant) {
                     Variant.ButtonsFunctional -> {
                         Subtitle(textRes = R.string.component_button_style_functional, horizontalPadding = true)
+                        val buttonStyles = listOf(OdsButton.Style.FunctionalPositive, OdsButton.Style.FunctionalNegative)
                         OdsChoiceChipsFlowRow(
-                            value = buttonStyle.value,
-                            onValueChange = { value -> buttonStyle.value = value },
+                            selectedChoiceChipIndex = buttonStyles.indexOf(buttonStyle.value),
                             modifier = Modifier.padding(horizontal = dimensionResource(id = com.orange.ods.R.dimen.spacing_m)),
-                            chips = listOf(
-                                OdsChoiceChip(
-                                    text = stringResource(id = R.string.component_button_style_functional_positive),
-                                    value = OdsButton.Style.FunctionalPositive
-                                ),
-                                OdsChoiceChip(
-                                    text = stringResource(id = R.string.component_button_style_functional_negative),
-                                    value = OdsButton.Style.FunctionalNegative
-                                )
-                            )
+                            choiceChips = buttonStyles.map { buttonStyle ->
+                                val text = when (buttonStyle) {
```

Comment:
I don't know what's better to read but what do you think about an `if`/`else` instead of a `when` in that case?